contracts/ForceMove.sol

```diff
@@ -704,14 +642,97 @@ contract ForceMove {
         // effects
 
         // set channel storage
-        channelStorageHashes[channelId] = keccak256(
-            abi.encode(ChannelStorage(0, now, bytes32(0), address(0), outcomeHash))
+        channelStorageHashes[channelId] = _getHash(
+            ChannelStorage(0, now, bytes32(0), address(0), outcomeHash)
         );
 
         // emit event
         emit Concluded(channelId);
     }
 
+    function _requireChannelOpen(uint48 turnNumRecord, bytes32 channelId) internal view {
+        // EITHER there is no information stored against channelId at all (OK)
+        // OR there is, in which case we must check the channel is still open
+        // and that the committed turnNumRecord is correct
+        require(
+            channelStorageHashes[channelId] == bytes32(0) ||
+                channelStorageHashes[channelId] ==
+                _getHash(ChannelStorage(turnNumRecord, 0, 0, address(0), 0)),
+            'Channel not open.'
+        );
+    }
+
+    function _requireMatchingStorage(ChannelStorage memory cs, bytes32 channelId) internal view {
+        require(
+            _matchesHash(cs, channelStorageHashes[channelId]),
+            'Channel storage does not match stored version.'
+        );
+    }
+
+    function _requireIncreasedTurnNumber(
+        ChannelStorage memory cs,
+        bytes32 channelId,
+        uint48 newTurnNumRecord
+    ) internal view {
+        _requireMatchingStorage(cs, channelId);
+        require(newTurnNumRecord > cs.turnNumRecord, 'turnNumRecord not increased.');
+    }
+
+    function _requireChallengePresent(ChannelStorage memory cs, bytes32 channelId) internal view {
+        _requireMatchingStorage(cs, channelId);
+        require(cs.finalizesAt > 0, 'Challenge not present.');
+    }
+
+    function _requireOngoingChallenge(ChannelStorage memory cs, bytes32 channelId) internal view {
+        _requireMatchingStorage(cs, channelId);
+        require(cs.finalizesAt > now, 'Challenge expired or not present.');
+
+    }
+
+    function _requireChannelNotFinalized(ChannelStorage memory cs, bytes32 channelId)
+        internal
+        view
+    {
+        require(cs.finalizesAt == 0 || cs.finalizesAt > now, 'Challenge expired');
+        _requireMatchingStorage(cs, channelId);
```

Comment:
Am I right in thinking the idea is to eventually remove this check on the storage from this function, but that that change-set is outside the scope of this PR? 

Perhaps we should open a new issue(s) to track the progress against the discussion in https://research.statechannels.org/t/channel-checkpointing-in-forcemove/24 .

Reply:
I see that #96 is in progress and will address this.